### PR TITLE
Cleanup footer

### PR DESCRIPTION
**Pull request by alexdgo**
Used loops to make footer cleaner. Added ts file with all the footer information.

### CODE REVIEW

File: src/components/Footer.tsx [hunk 1]
```diff
@@ -2,12 +2,18 @@ import React from 'react'
 import { Link } from 'gatsby'
 import { Row, Col, Image } from 'react-bootstrap'
 import s from 'styled-components'
+import { AboutLinks, NewsLinks, SportsLinks, OpinionLinks, StreetLinks, UnderTheButtonLinks } from '../constants'
 
 const StyledLink = s(Link)`
```
Comment (peterbaile) [inline src/components/Footer.tsx]: import `StyledLink` from `shared` folder

File: src/components/Footer.tsx [hunk 2]
```diff
@@ -2,12 +2,18 @@ import React from 'react'
...
   color: #000000 !important;
   text-decoration: none !important;
 `
 
+const StyledNonSlug = s.a`
```
Comment (peterbaile) [inline src/components/Footer.tsx]: rename it to `StyledAnchorTag`

File: src/components/Footer.tsx [hunk 3]
```diff
@@ -41,70 +47,82 @@ const Footer = () => (
...
-        <p> <StyledLink to="#"> Archives </StyledLink> </p>
-        <p> Staff </p>
-        <p> Advertise </p>
-        <p> Donate </p>
-        <p> Apparel Store </p>
-        <p> Alumni Association </p>
-        <p> Privacy Policy </p>
-        <p> Terms of Service </p>
-        <p> Contact us </p>
+        {AboutLinks.map((link, idx) => {
```
Comment (peterbaile) [inline src/components/Footer.tsx]: the same `if (idx === 0) return (...) else return (...)` appears multiple times, just modularize it into a single component and pass the `links` array to the component

**Review by peterbaile — CHANGES_REQUESTED** [review]
good job, but remove redundant code

**Comment by alexdgo** [comment]
Made functional components to make footer.tsx look nicer.
Moved components to shared folder for reusability.
Fixed anchor tags.

File: src/components/Footer.tsx [hunk 4]
```diff
@@ -27,85 +31,54 @@ const SNOImg = s.img`
...
+          return (
+            <p><StyledLink to={link.slug}>{link.section}</StyledLink></p>
+          )
+        }
+      })}
+    </Col>
+  )
+}
+
+const FooterAnchorLinks = linksArray => {
```
Comment (peterbaile) [inline src/components/Footer.tsx]: `FooterLinks` and `FooterAnchorLinks` can be merged into 1 component

**Review by peterbaile — CHANGES_REQUESTED** [review]
- merge the two components into one, and we are gtg

File: src/components/Footer.tsx [hunk 5]
```diff
@@ -27,85 +32,45 @@ const SNOImg = s.img`
...
-        <p> <StyledLink to="#"> Opinion </StyledLink> </p>
-        <p> <StyledLink to="#"> Video </StyledLink> </p>
-        <p> <StyledLink to="#"> Projects </StyledLink> </p>
-      </Col>
+        {FooterLinks(AboutLinks)}
+        {FooterLinks(NewsLinks)}
+        {FooterLinks(SportsLinks)}
+        {FooterLinks(OpinionLinks)}
+        {FooterLinks(StreetLinks)}
+        {FooterLinks(UnderTheButtonLinks)}
```
Comment (peterbaile) [inline src/components/Footer.tsx]: this should really be `<FooterLinks linksArray={UnderTheButtonLinks} />`

File: src/components/Footer.tsx [hunk 6]
```diff
@@ -27,85 +32,45 @@ const SNOImg = s.img`
   margin: 0 5px;
 `
 
-const Links = [
-  {
-    text: 'About',
-    slug: '/page/about',
-  }
-]
+const FooterLinks = (linksArray: ILink[]) => {
```
Comment (peterbaile) [inline src/components/Footer.tsx]: change this to proper interface type

**Review by peterbaile — CHANGES_REQUESTED** [review]
- formalize the way you write functional component

**Review by peterbaile — APPROVED** [review]

**Pull request merged**